src/communication/request_sync_send_queue.hpp

```diff
@@ -296,6 +298,18 @@ namespace cubcomm
     assert (backbuffer.empty ());
   }
 
+  template <typename ReqClient, typename ReqPayload>
+  void
+  request_sync_send_queue<ReqClient, ReqPayload>::wait_until_empty ()
```

Comment:
I think this function is incorrect as implemented here.
By looking at the implementation of `wait_not_empty_and_send_all` - specifically `m_request_queue.swap` - it does not guarantee that the return value is correct.
It would only work if there was a `m_shutdown` flag to instruct the `push` function to not accept requests anymore.
Or if you can guarantee that no more messages are posted to the send infrastructure.
Same remark for `request_sync_client_server::wait_until_all_requests_sent`.

Reply:
Agree.. It's sufficient for the use case in a unit test, but you're right. At least it needs a comment for this.